lib/game.rb

```diff
@@ -120,42 +124,22 @@ def switch_player
         @computer_player
     end
 
-    def win?
-      @players.each do |player|
-        if is_winner?(player)
-          return true
+    def declare_outcome(board, markers)
```

Comment:
You rely a lot on your `GameState` methods here - it might make sense to just have a method in `GameState` that gets the outcome for you internally and just passes it to `Game` when necessary